### PR TITLE
Feat/authentication/forgot password

### PR DESCRIPTION
**Pull request by hipe01mark**
## OVERVIEW

- Send reset password link via email
- Reset password via form request
- Reset password link activation

## SNIPPETS

image
image
image

### CODE REVIEW

File: app/Rules/EmailNotExistRule.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    /**
+     * Determine if the validation rule passes.
+     *
+     * @param  string  $attribute
+     * @param  mixed  $value
+     * @return bool
+     */
+    public function passes($attribute, $value)
+    {
+        if (User::whereEmail($value)->doesntExist()) {
```
Comment (mikco-hipe) [inline app/Rules/EmailNotExistRule.php]: suggestion 

maybe there is a method `exist()`? if so, the return is more optimized like ` return User::whereEmail($value)->exists();`

**Review by mikco-hipe — COMMENTED** [review]
One comment!

File: app/Rules/EmailNotExistRule.php (hunk 1 above)
Comment (hipe01mark) [reply app/Rules/EmailNotExistRule.php]: Done 🙇

**Review by karljerico — APPROVED** [review]
LGTM 🙇

**Review by mikco-hipe — APPROVED** [review]
LGTM 🙇

**Pull request merged**